upcloud/resource_upcloud_objectstorage.go

```diff
@@ -313,7 +315,18 @@ func copyObjectStorageDetails(objectDetails *upcloud.ObjectStorageDetails, d *sc
 		_ = d.Set("secret_key", "")
 	}
 
-	buckets, err := getBuckets(objectDetails.URL, accessKey, secretKey)
+	// This is an undocumented feature, can be changed or removed at any time without any announcements
+	// It allows you to also specify session token in env variables, which will be then passed to minio client
+	// along with access and secret key. This allows you to use temp credentials to read buckets list and such.
+	// This is only intended for import.
+	token := getToken(d)
```

Comment:
I kind of dislike that this checking for token and passing it to `getBucketConnection` is so implicit. It is very concise, but if the token is not set, if just passes empty string to `getBucketConnection` and there is no indication in the code that the lack of token was handled.
At the beginning I wanted to make those two things very separate in the code, like if there is a token set in env vars, we have a completely different code path. But that would duplicate a lot of code, and also gets a bit complicated with the fact that you totally can have secret key in env var, but access key in the config.
So at the end of the day I decided this ok-ish. It is implicit, but it's also very easy to remove if we ever want to do that

Reply:
What if we would change `getBucketConnection` signature to get `*upcloud.ObjectStorageDetails` as first argument e.g.
```
func getBucketConnection(objsto *upcloud.ObjectStorageDetails, accessKey, secretKey  string)
```
And them move that `getToken` inside that function and change it to `token := getToken(objsto.Name)`